src/caf/core/data_structures.py

```diff
@@ -684,8 +870,564 @@ def aggregate(self, segs: list[str]):
+                        "sums, so ipf will fail."
+                    )
+            # Check segmentations are compatible.
+            if not target.data.segmentation.is_subset(self.segmentation):
+                if target.segment_translations is None:
+                    raise ValueError(
+                        "The target segmentation is not a subset of the seed "
+                        "segmentation, but no correspondences are defined."
+                    )
+                non_matching = target.data.segmentation - self.segmentation
+                for seg in non_matching:
+                    seg = seg.name
+                    if seg in target.segment_translations.keys():
+                        if target.segment_translations[seg] in self.segmentation.names:
+                            lower_seg = self.segmentation.get_segment(
+                                target.segment_translations[seg]
+                            )
+                            try:
+                                # Don't need this for now, checking it exists.
+                                lower_seg.translate_segment(seg)
+                            except FileNotFoundError:
+                                raise FileNotFoundError(
+                                    f"No segment translation found for {lower_seg} to {seg}.from."
+                                )
+
+            # Check zoning systems are compatible.
+            if self.zoning_system != target.data.zoning_system:
+                zoning_diff = True
+                if target.zone_translation is not None:
+                    pass
+                else:
+                    try:
+                        target.zone_translation = self.zoning_system.translate(
+                            target.data.zoning_system
+                        )
+                    except TranslationError:
+                        raise TranslationError(
+                            "No zone_translation was found for "
+                            f"{self.zoning_system} to {target.data.zoning_system}."
+                        )
+                nested = (
+                    target.zone_translation[
+                        self.zoning_system.translation_column_name(target.data.zoning_system)
+                    ]
+                    == 1
+                ).all()
+                if not nested:
+                    raise TranslationError(
+                        "For IPF any targets must either be at the same zoning "
+                        "system as the seed DVector, or be at a zoning system "
+                        "which the seed nests perfectly within. The zone_translation "
+                        "found contains non-one factors, which implies the "
+                        "zoning system doesn't nest, so IPF can't be performed."
+                    )
+        return targets
+
+    def ipf(self, targets: list[IpfTarget], tol: float = 1e-5, max_iters: int = 100):
```

Comment:
Use `collections.abc.Collection` instead of list in parameter type annotations because it is less restrictive. Also, please could you add a return type annotation to help clarify outputs for users.